File: PodcastGPT.Frontend/models/dtos/PodcastSegment.ts
import { Podcast } from "./Podcast";
import { PodcastPersona } from "./PodcastPersona";

export interface PodcastSegment {
    podcastSegmentId: string;
    order: number;
    textContent: string;
    audioFileUrl: string;
    openAiRole: string;
    podcast: Podcast;
    podcastPersonaId: string;
    podcastPersona: PodcastPersona;
}

File: PodcastGPT.Data/Models/PodcastSegment.cs
namespace PodcastGPT.Data.Models;

public class PodcastSegment
{
	public Guid PodcastSegmentId { get; set; }
	public int Order { get; set; }
	public string TextContent { get; set; }
	public string? AudioFileUrl { get; set; }
	public string OpenAiRole { get; set; }

	public Podcast Podcast { get; set; }
	public Guid PodcastId { get; set; }
	
	public PodcastPersona PodcastPersona { get; set; }
	public Guid PodcastPersonaId { get; set; }
}